### PR TITLE
Add docs for new fixtures.

### PR DESCRIPTION
**Pull request by markstory**
This is likely not exhaustive but its a good start.

### CODE REVIEW

File: en/appendices/4-3-migration-guide.rst [hunk 1]
```diff
@@ -35,6 +35,12 @@ Log
 - ``SyslogLog`` moved the ``format`` config option to ``LegacySyslogFormatter``.
   Defaults to ``DefaultFormatter`` now.
 
+TestSuite
+---------
+
+- ``TestFixture::$fields`` and ``TestFixture::$import`` are deprecated you
```
Comment (othercorey) [inline en/appendices/4-3-migration-guide.rst]:
```suggestion
- ``TestFixture::$fields`` and ``TestFixture::$import`` are deprecated. You
```

File: en/appendices/4-3-migration-guide.rst [hunk 2]
```diff
@@ -35,6 +35,12 @@ Log
...
+  should convert your application to use the :doc:`new fixture system <./fixture-upgrade>`.
```
Comment (othercorey) [inline en/appendices/4-3-migration-guide.rst]:
```suggestion
  should convert your application to the :doc:`new fixture system <./fixture-upgrade>`.
```

File: en/appendices/fixture-upgrade.rst [hunk 3]
```diff
@@ -0,0 +1,146 @@
+Fixture Upgrade
+###############
+
+As of 4.3.0 a fixture schema and data management responsibilities were split.
```
Comment (othercorey) [inline en/appendices/fixture-upgrade.rst]:
```suggestion
As of 4.3.0 fixture schema and data management responsibilities were split.
```

File: en/appendices/fixture-upgrade.rst [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+Maintaining schema in fixture classes and migrations added complexity and
+maintenance cost to applications. In 4.3.0 new APIs were added to enable you to
+more easily re-use your existing migrations or schema management tools with
+tests.
+
+To upgrade to the new fixture system, you need to make a few updates:
+
+#. First open your ``phpunit.xml`` file and remove the ``<listeners>`` block.
```
Comment (othercorey) [inline en/appendices/fixture-upgrade.rst]:
```suggestion
#. First, remove the ``<listeners>`` block from your ``phpunit.xml``.
```

File: en/appendices/fixture-upgrade.rst [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+#. In ``phpunit.xml`` add the followin::
```
Comment (othercorey) [inline en/appendices/fixture-upgrade.rst]:
```suggestion
#. Add the following to your ``phpunit.xml``::
```

File: en/appendices/fixture-upgrade.rst [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+
+    <extensions>
+        <extension class="\Cake\TestSuite\FixtureSchemaExtension" />
+    </extensions>
```
Comment (othercorey) [inline en/appendices/fixture-upgrade.rst]:
```suggestion
        <extensions>
            <extension class="\Cake\TestSuite\FixtureSchemaExtension" />
        </extensions>
```

File: en/appendices/fixture-upgrade.rst [hunk 7]
```diff
@@ -0,0 +1,146 @@
...
+   This removes schema management from the test fixture manager. Instead your
+   application needs to create/update schema at the beginning of each test run.
+#. Next update ``tests/boostrap.php`` to create schema. There are a few
+   different ways to create schema. Refer to :ref:`creating-test-database-schema`
+   for the methods provided by CakePHP.
+#. Then, remove all the ``$fields`` and ``$import`` properties from your fixtures.
+   These properties are unused in the new fixture system.
+
+Your tests should continue to pass, and you can experiment with
+:ref:`fixture-state-managers`. The ``TransactionStrategy`` can yield significant
```
Comment (othercorey) [inline en/appendices/fixture-upgrade.rst]:
```suggestion
:ref:`fixture-state-managers`. ``TransactionStrategy`` can yield significant
```

File: en/appendices/fixture-upgrade.rst [hunk 8]
```diff
@@ -0,0 +1,146 @@
...
+performance improvements.
+
+Legacy Fixture Documentation
+================================
+
+The following documentation applies only to the listener based fixtures that are
```
Comment (othercorey) [inline en/appendices/fixture-upgrade.rst]:
```suggestion
The following documentation applies only to the listener-based fixtures that are
```

File: en/appendices/fixture-upgrade.rst [hunk 9]
```diff
@@ -0,0 +1,146 @@
...
+#. First open your ``phpunit.xml`` file and remove the ``<listeners>`` block.
+#. In ``phpunit.xml`` add the followin::
+
+    <extensions>
+        <extension class="\Cake\TestSuite\FixtureSchemaExtension" />
+    </extensions>
+
+   This removes schema management from the test fixture manager. Instead your
+   application needs to create/update schema at the beginning of each test run.
+#. Next update ``tests/boostrap.php`` to create schema. There are a few
```
Comment (othercorey) [inline en/appendices/fixture-upgrade.rst]:
```suggestion
#. Next, update ``tests/boostrap.php`` to create schema. There are a few
```

File: en/appendices/fixture-upgrade.rst [hunk 10]
```diff
@@ -0,0 +1,146 @@
...
+Let's start with an example. Assuming you have a table named articles available
+in your application, change the example fixture given in the previous section
+(**tests/Fixture/ArticlesFixture.php**) to::
+
+    class ArticlesFixture extends TestFixture
+    {
+        public $import = ['table' => 'articles'];
+    }
+
+If you want to use a different connection use::
```
Comment (othercorey) [inline en/appendices/fixture-upgrade.rst]:
```suggestion
If you want to use a different connection, use::
```

File: en/appendices/fixture-upgrade.rst [hunk 11]
```diff
@@ -0,0 +1,146 @@
...
+
+Defining the schema in fixture files can be really handy when creating plugins
+or libraries or if you are creating an application that needs to be portable
+between database vendors. Redefining the schema in fixtures can become difficult
+to maintain in larger applications. Because of this CakePHP provides the ability
+to import the schema from an existing connection and use the reflected table
+definition to create the table definition used in the test suite.
+
+Let's start with an example. Assuming you have a table named articles available
+in your application, change the example fixture given in the previous section
```
Comment (othercorey) [inline en/appendices/fixture-upgrade.rst]:
```suggestion
Let's start with an example. Assuming you have a table named articles, change the example
fixture given in the previous section
```

File: en/appendices/fixture-upgrade.rst [hunk 12]
```diff
@@ -0,0 +1,146 @@
...
+    }
+
+If you want to use a different connection use::
+
+    class ArticlesFixture extends TestFixture
+    {
+        public $import = ['table' => 'articles', 'connection' => 'other'];
+    }
+
+Usually, you have a Table class along with your fixture, as well. You can also
```
Comment (othercorey) [inline en/appendices/fixture-upgrade.rst]:
```suggestion
Usually, you have a Table class along with your fixture. You can also
```

**Comment by othercorey** [comment]
warning from build:

> /home/runner/work/docs/docs/en/appendices/fixture-upgrade.rst: WARNING: document isn't included in any toctree

> /home/runner/work/docs/docs/en/appendices/fixture-upgrade.rst:27: WARNING: undefined label: fixture-state-managers (if the link has no caption the label must precede a section heade

**Review by Arhell — APPROVED** [review]

**Pull request merged**